domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.cs

```diff
@@ -0,0 +1,167 @@
+    {
+        var idpUserId = await SeedData();
+
+        var userClient = _integrationTestFixture.WebAppFactory.CreateApi(sub: idpUserId.Value.ToString());
+
+        var response = await userClient.GetUserOrganizationConsents();
+
+        response.Should().Be200Ok();
+
+        var result = await response.Content.ReadFromJsonAsync<GetUserOrganizationConsentsQueryResult>();
+        result!.Result.Should().NotBeEmpty();
+    }
+
+    [Fact]
+    public async Task GivenUser_WhenGettingConsent_ThenVerifyResponseSnapshot()
+    {
+        var idpUserId = await SeedData();
+        var userClient = _integrationTestFixture.WebAppFactory.CreateApi(sub: idpUserId.Value.ToString());
+        var response = await userClient.GetUserOrganizationConsents();
+
+        response.Should().Be200Ok();
+
+        var getConsentAgreement = JsonConvert.DeserializeObject<GetUserOrganizationConsentsQueryResult>(await response.Content.ReadAsStringAsync());
+
+        var settings = new VerifySettings();
+        settings.ScrubMembersWithType(typeof(long));
+
+        await Verify(getConsentAgreement, settings);
+    }
+
+    [Fact]
+    public async Task GivenNoConsentsExist_WhenUserQueriesForConsents_HttpOKEmptyBody()
+    {
+        await SeedData();
+
+        var response = await _api.GetUserOrganizationConsents();
+
+        response.Should().Be200Ok();
+
+        var deserializedResponse = JsonConvert.DeserializeObject<GetUserOrganizationConsentsQueryResult>(await response.Content.ReadAsStringAsync());
+
+        await Verify(deserializedResponse);
+    }
+
+    [Fact]
+    public async Task GivenUserAffiliatedWithMultipleOrganizations_WhenGettingConsent_ThenOnlyConsentFromCurrentOrganizationContextIncludedInResponse()
+    {
+        var user = Any.User();
+        var organization1 = Any.Organization();
+        var organization2 = Any.Organization(Tin.Create("87654321"));
+        var client1 = Any.Client();
+        var client2 = Any.Client();
+
+        var consent1 = Consent.Create(organization1, client1, DateTimeOffset.UtcNow);
+        var consent2 = Consent.Create(organization2, client2, DateTimeOffset.UtcNow);
+
+        await using var dbContext = new ApplicationDbContext(_options);
+        await dbContext.Users.AddAsync(user);
+        await dbContext.Organizations.AddRangeAsync([organization1, organization2]);
+        await dbContext.Clients.AddRangeAsync([client1, client2]);
+
+        var affiliation1 = Affiliation.Create(user, organization1);
+        var affiliation2 = Affiliation.Create(user, organization2);
+
+        await dbContext.Affiliations.AddRangeAsync([affiliation1, affiliation2]);
+        await dbContext.Consents.AddRangeAsync([consent1, consent2]);
+
+        await dbContext.SaveChangesAsync();
+
+        var userIdString = user.IdpUserId.Value.ToString();
+
+        var userClient = _integrationTestFixture.WebAppFactory.CreateApi(sub: userIdString);
+        var response = await userClient.GetUserOrganizationConsents();
```

Comment:
This and the check for 200 OK can be combined into a single GetFromJsonAsync method call.